src/org/ods/orchestration/util/MROPipelineUtil.groovy

```diff
@@ -324,6 +326,14 @@ class MROPipelineUtil extends PipelineUtil {
                         } else {
                             this.logger.debug("Repo '${repo.id}' is of type ODS Code Component. Nothing to do in phase '${name}' for target environment '${targetEnvToken}'.")
                         }
+                    } else if (repo.type?.toLowerCase() == PipelineConfig.REPO_TYPE_ODS_INFRA) {
+                        if (this.project.isAssembleMode && name == PipelinePhases.BUILD) {
+                            executeODSComponent(repo, baseDir)
+                        } else if (this.project.isPromotionMode && name == PipelinePhases.BUILD) {
```

Comment:
@clemensutschig `ods` repo types deploy to Q/P in `Phase.DEPLOY`. Won't a promotion in `Phase.BUILD` rule out orchestration of code and infra components?